samples/sqltransport/multi-instance-migration/SqlTransport_4/Receiver.V3/Program.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Threading.Tasks;
+using Messages;
+using NServiceBus;
+using NServiceBus.Transport.SQLServer;
+#pragma warning disable 618
+
+class Program
+{
+
+    static void Main()
+    {
+        AsyncMain().GetAwaiter().GetResult();
+    }
+
+    static async Task AsyncMain()
+    {
+        Console.Title = "Samples.SqlServer.MultiInstanceReceiver.V3";
+
+        #region ReceiverConfigurationV3
+
+        var endpointConfiguration = new EndpointConfiguration("Samples.SqlServer.MultiInstanceReceiver");
+        var transport = endpointConfiguration.UseTransport<SqlServerTransport>();
+        transport.EnableLegacyMultiInstanceMode(ConnectionProvider.GetConnection);
+        endpointConfiguration.UsePersistence<InMemoryPersistence>();
+        endpointConfiguration.SendFailedMessagesTo("error");
+        endpointConfiguration.EnableInstallers();
+
+        #endregion
+
+        SqlHelper.EnsureDatabaseExists(ConnectionProvider.DefaultConnectionString);
+
+        endpointConfiguration.Conventions().DefiningMessagesAs(t => t.Assembly == typeof(ClientOrder).Assembly && t.Namespace == "Messages");
```

Comment:
why do we need conventions?

Reply:
Yes because the messages assembly is shared between V3 and V4 versions.